lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -173,14 +369,25 @@ try
   $ps.Invoke()
 
   [Void]$ps.AddScript($powershell_code)
+  $ps.Commands.Commands[0].MergeMyResults(
+    [System.Management.Automation.Runspaces.PipelineResultTypes]::All,
+    [System.Management.Automation.Runspaces.PipelineResultTypes]::Output);
+  [void]$ps.AddCommand("out-default");
+
   $asyncResult = $ps.BeginInvoke()
 
   if (!$asyncResult.AsyncWaitHandle.WaitOne(<%= timeout_ms %>, $false)){
     throw "Catastrophic failure: PowerShell module timeout (<%= timeout_ms %> ms) exceeded while executing"
   }
 
   $output = $ps.EndInvoke($asyncResult)
-  $output = $output | Out-String
+  #$output = $output | Out-String
+  #if($ps.HasErrors){
+  #  $ps.Streams.Error.ReadAll() | %{
+  #    $puppetPSHost.UI.WriteLine($_.ToString())
+  #  }
+  #}
+  $output = [string]($puppetPSHost.UI.Output)
```

Comment:
Since we are getting out output from the UI we don't actually need the output returned from the Invoke() call, as that will always be empty because of the host we implemented